server/views/partials/expectedArrivalCardMacro.njk

```diff
@@ -12,7 +12,7 @@
             <div class="app-card__content">
                 <header class="app-card__header govuk-!-margin-bottom-4">
                     <h2 class="app-card__title" data-qa="{{moveType}}-title-{{loopIndex}}">
-                        {% if arrival.fromLocationType === 'COURT' and not arrival.prisonNumber %}
+                        {% if arrival.fromLocationType === 'COURT' and not arrival.isCurrentPrisoner %}
```

Comment:
So from court we have:

- Matched & out of prison = Has prisonNumber and&isCurrentPrisoner is false = show link
- Matched & In prison = Has prisonNumber & isCurrentPrisoner  is true = no link
- Unmatched = No prisonNumber &  isCurrentPrisoner is false = show link

think the above logic covers all these three scenarios